services/src/main/java/org/keycloak/protocol/oidc/endpoints/TokenEndpoint.java

```diff
@@ -743,6 +744,13 @@ public Response clientCredentialsGrant() {
 
         event.success();
 
+        // KEYCLOAK-9551 Client Credentials Grant generates refresh token handling
+        if (!useRefreshToken) {
```

Comment:
So the other PR for "transient sessions" was merged before yours, see the commit https://github.com/keycloak/keycloak/commit/ff05072c164d95e5084fba7d556820d1ccbe7f31 . So I suggest that you rebase to latest master and do:
- When calling "createUserSession" in TokenEndpoint.clientCredentialsGrant(), you can add the flag PERSISTENT or TRANSIENT as last parameter to the method based on if refreshToken is required. This will allow to remove TODO I've added in the other PR. Also it won't be needed to call this "removeUserSession", so it can be removed entirely.

Reply:
Thanks for the feedback!

I'll give this a shot at the weekend.